imageregistry/v1/types.go

```diff
@@ -306,6 +306,81 @@ type ImageRegistryConfigStorageIBMCOS struct {
+	PlainText AlibabaEncryptionMode = "PlainText"
+	// AES256 is an AlibabaEncryptionMode. This means AES256 encryption
+	AES256 AlibabaEncryptionMode = "AES256"
+	// KMS is an AlibabaEncryptionMode. This means KMS encryption
+	KMS AlibabaEncryptionMode = "KMS"
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	// EncryptionType defines the different encrytion types available
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `PlainText`.
+	// +kubebuilder:validation:Enum="PlainText";"KMS";"AES256"
+	// +kubebuilder:default="PlainText"
+	// +kubebuilder:validation:Required
+	Type AlibabaEncryptionMode `json:"type"`
+
+	// KMS (key management service) is an encryption type that holds the struct for KMS KeyID
+	// +optional
+	KMS *KMSEncryptionAlibaba `json:"kms,omitempty"`
+}
+
+type KMSEncryptionAlibaba struct {
+	// KeyID holds the KMS encryption key ID
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	KeyID string `json:"keyID"`
+}
+
+// ImageRegistryConfigStorageAlibabaOSS holds Alibaba Cloud OSS configuration.
+// Configures the registry to use Alibaba Cloud Object Storage Service for backend storage.
+// More about oss, you can look at the [official documentation](https://www.alibabacloud.com/help/product/31815.htm)
+type ImageRegistryConfigStorageAlibabaOSS struct {
+	// Bucket is the bucket name in which you want to store the registry's data.
+	// About Bucket naming, more details you can look at the [official documentation](https://www.alibabacloud.com/help/doc-detail/257087.htm)
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default will be autogenerated in the form of <clusterid>-image-registry-<region>-<random string 27 chars>
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:MinLength=3
+	// +kubebuilder:validation:Pattern=`^[0-9a-z]+(-[0-9a-z]+)*$`
+	// +optional
+	Bucket string `json:"bucket,omitempty"`
+	// Region is the Alibaba Cloud Region in which your bucket exists.
+	// For a list of regions, you can look at the [official documentation](https://www.alibabacloud.com/help/doc-detail/31837.html).
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default will be based on the installed Alibaba Cloud Region.
+	// +optional
+	Region string `json:"region,omitempty"`
+	// EndpointAccessibility specifies whether the registry use the OSS VPC internal endpoint
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `Internal`.
+	// +kubebuilder:validation:Enum="Internal";"Public"
```

Comment:
also need `""`, because client side validation does not see the default.

Reply:
Will add.